File: JAVA/SRobotics/src/sRobotics/core/state/ActivationState.java
package sRobotics.core.state;

import sRobotics.core.RobotContext;

public class ActivationState implements IState<ActivationStateType>
{
    private ActivationStateType activationState;

    @Override
    public void Change(RobotContext context, ActivationStateType activationState) {
        this.activationState = activationState;

        hello(context);
    }

    private void hello(RobotContext context)
    {
        context.getCommands().sayHello.Execute(context);
    }
}

File: C#/SRobotics/Core/State/ActivationState.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.State
{
    public enum ActivationStateType
    {
        Off,
        On
    }

    public class ActivationState : IState<ActivationStateType>
    {
        private ActivationStateType activationState;

        /*public void ChangeActivationState(ActivationStateType state)
        {
            activationState = state;
        }*/

        public void Change(ActivationStateType value)
        {
            activationState = value;
        }
    }
}
